From 063120d73fce79b618e85cdb1f8a5379c325b6a3 Mon Sep 17 00:00:00 2001
From: Peter Qiu <zqiu@chromium.org>
Date: Fri, 21 Aug 2015 10:44:13 -0700
Subject: [PATCH] DHCPv6: remove ia_na request

Currently, the main use case for DHCPv6 is for Prefix Delegation over
PPPoE connection. The DHCPv6 server is usually not configured to provide
IA_NA option.  So disable this request for now.

TODO: add support to specify DHCPv6 request options over command line.

BUG=chromium:523196
TEST=USE=ipv6 emerge-$BOARD dhcpcd
TEST=Run network_Dhcpv6Basic test
---
 if-options.c | 20 +++++---------------
 1 file changed, 5 insertions(+), 15 deletions(-)

diff --git a/if-options.c b/if-options.c
index 1c0129d..d5d91a2 100644
--- a/if-options.c
+++ b/if-options.c
@@ -676,30 +676,20 @@ parse_option(struct dhcpcd_ctx *ctx, const char *ifname, struct if_options *ifo,
 #ifdef INET6
 	case 'a':
 		/* Chromeos hack: configure DHCPv6 option for shill. */
+		/* TODO(zqiu): make request options configurable. */
 
-		/* Reallocate ia to add both ia_na and ia_pd. */
-		ia = realloc(ifo->ia, sizeof(*ifo->ia) * (ifo->ia_len + 2));
+		/* Reallocate ia to add account for ia_pd option. */
+		ia = realloc(ifo->ia, sizeof(*ifo->ia) * (ifo->ia_len + 1));
 		if (ia == NULL) {
 			logger(ctx, LOG_ERR, "%s: %m", __func__);
 			return -1;
 		}
 		ifo->ia = ia;
 
-		/* Setup ia_na option with iaid of 0. */
-		ia = &ifo->ia[ifo->ia_len++];
-		ia->ia_type = D6_OPTION_IA_NA;
-		parse_iaid(ia->iaid, "0", sizeof(ia->iaid));
-		ia->iaid_set = 1;
-		memset(&ia->addr, 0, sizeof(ia->addr));
-		ia->prefix_len = 0;
-		ia->sla_max = 0;
-		ia->sla_len = 0;
-		ia->sla = NULL;
-
-		/* Setup ia_pd option with iaid of 1. */
+		/* Setup ia_pd option with iaid of 0. */
 		ia = &ifo->ia[ifo->ia_len++];
 		ia->ia_type = D6_OPTION_IA_PD;
-		parse_iaid(ia->iaid, "1", sizeof(ia->iaid));
+		parse_iaid(ia->iaid, "0", sizeof(ia->iaid));
 		ia->iaid_set = 1;
 		memset(&ia->addr, 0, sizeof(ia->addr));
 		ia->prefix_len = 0;
-- 
2.2.0.rc0.207.ga3a616c

